### DISCUSSION TITLE
Network Error

### DISCUSSION THREAD

**Discussion by its-khushi** [otahina/PowerPoint-Generator-Python-Project#13] [Q&A]
While clicking on generate ppt I am getting this error:
![image](https://github.com/otahina/PowerPoint-Generator-Python-Project/assets/94925949/f82da1c0-526f-438a-8665-7ab6e355f1b9)
How can I resolve this error?

**Comment by manasi-gade07** [comment]
# Fix for 500 Internal Server Error

The error happens because no template is selected, and the code breaks.
Update your JS like this:

```js
const selectedTemplate = document.querySelector('input[name="template_choice"]:checked');
if (!selectedTemplate) {
  alert("Please select a template");
  return;
}
formData.append('template_choice', selectedTemplate.value);
```

This makes sure the value is only added if a template is selected, avoiding the crash.
